src/GitVersionExe/GitVersionApp.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace GitVersion
+{
+    internal class GitVersionApp : IHostedService
+    {
+        private readonly IHostApplicationLifetime applicationLifetime;
+        private readonly IGitVersionRunner gitVersionRunner;
+        private readonly Arguments arguments;
+
+        public GitVersionApp(IHostApplicationLifetime applicationLifetime, IGitVersionRunner gitVersionRunner, IOptions<Arguments> options)
+        {
+            arguments = options.Value;
```

Comment:
@arturcic is there a reason that this doesn't have a `this.`, similar to the other two?